great_expectations/datasource/fluent/spark_filesystem_datasource.py

```diff
@@ -56,20 +65,36 @@ def test_connection(self, test_assets: bool = True) -> None:
                 asset.test_connection()
 
     def _build_data_connector(
-        self, data_asset: CSVAsset, glob_directive: str = "**/*", **kwargs
+        self,
+        data_asset: _SPARK_FILE_PATH_ASSET_TYPES,
+        glob_directive: str = "**/*",
+        **kwargs,
     ) -> None:
         """Builds and attaches the `FilesystemDataConnector` to the asset."""
         if kwargs:
             raise TypeError(
                 f"_build_data_connector() got unexpected keyword arguments {list(kwargs.keys())}"
             )
+        if isinstance(data_asset, _SPARK_DIRECTORY_ASSET_CLASSES):
+            get_unfiltered_batch_definition_list_fn = cast(
+                Callable[[FilePathDataConnector, BatchRequest], List[BatchDefinition]],
+                partial(
+                    directory_get_unfiltered_batch_definition_list_fn,
+                    data_directory=data_asset.data_directory,
+                ),
+            )
```

Comment:
I was also considering using a Protocol here for the function partial (e.g. https://stackoverflow.com/a/66664521/2916375) but open to ideas.

Reply:
Oh I didn't know `partials` don't return nice `callables` like we want. I think the cleanest solution might be sidestepping the use of partials and instead replace the definition of `directory_get_unfiltered_batch_definition_list_fn` with something like:
```
def make_directory_get_unfiltered_batch_definition_list_fn(data_directory: str) -> Callable[[FilePathDataConnector, BatchRequest], list[BatchDefinition]]:

    def directory_get_unfiltered_batch_definition_list_fn(data_connector: FilePathDataConnector, batch_request: BatchRequest):
        batch_definition_list: List[BatchDefinition] = list()
        batch_definition = BatchDefinition(
            datasource_name=data_connector.datasource_name,
            data_connector_name=_DATA_CONNECTOR_NAME,
            data_asset_name=data_connector.data_asset_name,
            batch_identifiers=IDDict({"path": data_directory}),
        )
        batch_definition_list.append(batch_definition)

        return [batch_definition]

    return directory_get_unfiltered_batch_definition_list_fn
```
Then we could make this line:
```
get_unfiltered_batch_definition_list_fn = make_directory_get_unfiltered_batch_definition_list_fn(data_asset.data_directory)
```

Reply:
Good call, done!